source/renderer/app/components/staking/delegation-center/UndelegateConfirmationDialog.js

```diff
@@ -0,0 +1,268 @@
+import { submitOnEnter } from '../../../utils/form';
+
+const messages = defineMessages({
+  dialogTitle: {
+    id: 'staking.delegationCenter.undelegate.dialog.title',
+    defaultMessage: '!!!Undelegate',
+    description: 'Title for the "Undelegate" dialog.',
+  },
+  confirmButtonLabel: {
+    id: 'staking.delegationCenter.undelegate.dialog.confirmButtonLabel',
+    defaultMessage: '!!!Undelegate',
+    description:
+      'Label for the "Undelegate" button in the "Undelegate" dialog.',
+  },
+  description: {
+    id: 'staking.delegationCenter.undelegate.dialog.description',
+    defaultMessage:
+      '!!!<p>The stake from your wallet <strong>{walletName}</strong> is currently delegated to the <strong>[{stakePoolSlug}] {stakePoolName}</strong> stake pool.</p><p>Do you want to undelegate your stake and stop earning rewards?</p>',
+    description: 'Description for the "Undelegate" dialog.',
+  },
+  confirmUnsupportCheck: {
+    id: 'staking.delegationCenter.undelegate.dialog.confirmUnsupportCheck',
+    defaultMessage:
+      '!!!I understand that I am not supporting the Cardano network when my stake is undelegated.',
+    description:
+      'Label for the unsupport confirmation check in the "Undelegate" dialog.',
+  },
+  confirmUneligibleCheck: {
+    id: 'staking.delegationCenter.undelegate.dialog.confirmUneligibleCheck',
+    defaultMessage:
+      '!!!I understand that I will not be eligible to earn rewards when my stake is undelegated.',
+    description:
+      'Label for the uneligible confirmation check in the "Undelegate" dialog.',
+  },
+  feesLabel: {
+    id: 'staking.delegationCenter.undelegate.dialog.feesLabel',
+    defaultMessage: '!!!Fees',
+    description: 'Fees label in the "Undelegate" dialog.',
+  },
+  adaLabel: {
+    id: 'staking.delegationCenter.undelegate.dialog.adaLabel',
+    defaultMessage: '!!!ADA',
+    description: 'ADA label in the "Undelegate" dialog.',
+  },
+  spendingPasswordLabel: {
+    id: 'staking.delegationCenter.undelegate.dialog.spendingPasswordLabel',
+    defaultMessage: '!!!Spending password',
+    description: 'Spending password label in the "Undelegate" dialog.',
+  },
+  spendingPasswordPlaceholder: {
+    id:
+      'staking.delegationCenter.undelegate.dialog.spendingPasswordPlaceholder',
+    defaultMessage: '!!!Type your spending password here',
+    description: 'Spending password placeholder in the "Undelegate" dialog.',
+  },
+  passwordErrorMessage: {
+    id: 'staking.delegationCenter.undelegate.dialog.passwordError',
+    defaultMessage: '!!!Incorrect spending password.',
+    description: 'Label for password error in the "Undelegate" dialog.',
+  },
+});
+
+type Props = {
+  walletName: string,
+  stakePoolName: string,
+  stakePoolSlug: string,
+  onConfirm: Function,
+  onCancel: Function,
+  onExternalLinkClick: Function,
+  isSubmitting: boolean,
+  error: ?LocalizableError,
+  fees: BigNumber,
+};
+
+type State = {
```

Comment:
Isn't better to use `ReactToolboxMobxForm` instead of State? It's good for consistency, also improves the UX.

Reply:
@daniloprates This was fixed. Thanks.